# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaSpan.h
# Commit: f53080eb8868
# Full Hash: f53080eb8868a5031861a0182433701dae21f1dd
# Author: Chris Pearce <cpearce@mozilla.com>
# Date: 2019-06-14 10:00:22
# Regressor Bug: 1558364
# File Overlap Count: 1
# Description:
#   Bug 1558364 - Convert TrackBuffersManager::mPendingInput into a MediaSpan. r=jya
#   
#   This allows us to avoid a (probably small) copy when we stash the pending input.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D34662
# ==============================================================================

diff -r f6df4eb84932 -r f53080eb8868 dom/media/MediaSpan.h
--- a/dom/media/MediaSpan.h	Fri Jun 14 02:10:09 2019 +0000
+++ b/dom/media/MediaSpan.h	Fri Jun 14 02:10:24 2019 +0000
@@ -64,6 +64,8 @@
     return (*mBuffer)[mStart + aIndex];
   }
 
+  bool Append(const MediaSpan& aBuffer) { return Append(aBuffer.Buffer()); }
+
   bool Append(MediaByteBuffer* aBuffer) {
     if (!aBuffer) {
       return true;